analyzer/codechecker_analyzer/analysis_manager.py

```diff
@@ -647,10 +665,16 @@ def __create_timeout(analyzer_process):
                     LOG.error("Analyzing '%s' with %s without CTU failed.",
                               source_file_name, action.analyzer_type)
 
-                    zip_file = result_base + '.zip'
-                    zip_file = os.path.join(failed_dir, zip_file)
-                    handle_failure(source_analyzer, rh, zip_file,
-                                   result_base, actions_map)
+                    if generate_reproducer:
+                        handle_reproducer(
+                            source_analyzer, rh,
+                            os.path.join(reproducer_dir, result_base + '.zip'),
+                            actions_map)
+                    else:
+                        handle_failure(
+                            source_analyzer, rh,
+                            os.path.join(failed_dir, result_base + '.zip'),
+                            result_base, actions_map)
```

Comment:
This pattern is repeated a couple of times. Maybe we can create a separate function or something similar for it:
```py
def handle_analysis_(success, generate_reproducer, ...):
    """ TODO """
    if success:
        handle_success()

    if generate_reproducer:
        handle_reproducer()
    elsif not success:
        handle_failure()
```
And we can call this function at least 4 places.

Reply:
I can't see the advantage of this extra function. This would require a function with as many parameters as `handle_success()`, `handle_failure()` and `handle_reproducer()` have altogether, plus another which indicates whether it was a success or failure. Personally I think it doesn't help readability to introduce one more handler function where we have to follow through how to handle the result of an analysis. What do you think?

Reply:
Yes, you are right, this function would contain a lot of parameters:
```py
def handle_analysis_return(success, generate_reproducer, source_analyzer,
                           rh, zip_file, actions_map, result_base, skip_handler,
                           capture_analysis_output, success_dir):
    ...
```

But what if we create an inner function inside this check function:
```py
def check():
    source_analyzer = ...
    rh = ...

    def handle_analysis_return(success, zip_file):
        """ MY HELPER FUNCTION """
        pass

    if rh.analyzer_returncode == 0:
        handle_analysis_return(True, reproducer_zip_file)
```
What do you think about this solution? I just want to avoid so much code duplications.

Reply:
Done